lib/string_basics.dart

```diff
@@ -200,6 +201,46 @@ extension StringBasics on String {
     }
     return stringBuffer.toString();
   }
+
+  /// Returns a truncated version of the string.
+  ///
+  /// Example:
+  /// ```dart
+  /// final sentence = 'The quick brown fox jumps over the lazy dog';
+  /// final truncated = sentence.truncate(20); // 'The quick brown fox...'
+  /// ```
+  ///
+  /// The [length] is the truncated length of the string.
+  /// The [substitution] is the substituting string of the truncated characters.
+  /// If not null or empty it will be appended at the end of the truncated string.
+  /// The [trimTrailingWhitespace] is whether or not to trim the spaces of the truncated string
+  /// before appending the ending string.
+  /// The [includeSubstitutionLength] is whether or not that the length of the substitution string will be included
+  /// with the intended truncated length.
+  String truncate(
+    int length, {
+    String substitution = '',
+    bool trimTrailingWhitespace = true,
+    bool includeSubstitutionLength = false,
+  }) {
+    if (this.length <= length) {
+      return this;
+    }
+
+    String truncated = this.characters.take(length).toString();
+
+    if (includeSubstitutionLength) {
+      // reduce the length with substitution length
+      final newLength = truncated.length - substitution.length;
+
+      // reduce the truncated string by the substitution length
+      // since we will be considering the substitution to the target length
+      truncated = truncated.characters.take(newLength).toString();
+    }
```

Comment:
Could this be simplified with something like:

```
final truncatedLength = includeSubstitituionLength ? (length - substitution.length) : length;
final truncated = this.characters.take(truncatedLength).toString();
```

?

Reply:
Yeah, that is way better. I updated it. Thanks.